### PR TITLE
Update ruff and add pre-commit

### PR DESCRIPTION
**Pull request by javierggt**
## Description

Update to the latest ruff and add pre-commit.

### Unit tests

- [ ] No unit tests
- [ ] Mac
- [ ] Linux
- [ ] Windows

Independent check of unit tests by [REVIEWER NAME]
- [ ] [PLATFORM]:

### Functional tests

No functional testing.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+repos:
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.1.5
```
Comment (taldcroft) [inline .pre-commit-config.yaml:4]: Should be 0.2.1.

**Review by taldcroft — COMMENTED** [review]
Thanks, looks good except the pre-commit version number.

**Comment by javierggt** [comment]
Fixed.

**Review by taldcroft — APPROVED** [review]
LGTM

**Pull request merged**